### PR TITLE
Implemented notifications from github API

### PR DESCRIPTION
**Pull request by dennisdegreef**

### CODE REVIEW

File: lib/Github/Client.php [hunk 1]
```diff
@@ -143,6 +145,11 @@ public function api($name)
                 $api = new Api\Markdown($this);
                 break;
 
+            case 'notification':
+            case 'notifications':
+                $api = new Api\Activity\Notification($this);
```
Comment (stof) [inline lib/Github/Client.php]: This is wrong. If it is in the ``Activity`` subnamespace, it should be acess from an ``Api\Activity`` API, not from the main client, to be consistent with the current architecture of the library

Comment (dennisdegreef) [inline lib/Github/Client.php]: Moved to namespace `Api\Notification` see `300a091`

File: lib/Github/Api/Notification.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    /**
+     * Marks all notifications as read from the current date
+     * Optionally give DateTimeInterface to mark as read before that date
+     *
+     * @link https://developer.github.com/v3/activity/notifications/#mark-as-read
+     *
+     * @param DateTimeInterface $since
+     *
+     */
+    public function markRead(DateTimeInterface $since = null)
```
Comment (stof) [inline lib/Github/Api/Notification.php]: typehinting the interface means that it is not compatible with PHP 5.3 or 5.4

**Comment by stof** [comment]
Tests are missing

**Comment by dennisdegreef** [comment]
Added test and fixed PHP5.3 and 5.4 compatibility.

File: lib/Github/Api/Notification.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+ *
+ * Important! You have to be authenticated to perform these methods
+ *
+ * @link   https://developer.github.com/v3/activity/notifications/
+ * @author Dennis de Greef <github@link0.net>
+ */
+class Notification extends AbstractApi
+{
+    /**
+     * Get a listing of a notifications
```
Comment (xabbuh) [inline lib/Github/Api/Notification.php]: The `a` should be removed here.

File: lib/Github/Api/Notification.php [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+     * @link https://developer.github.com/v3/activity/notifications/
+     *
+     * @param bool $includingRead
+     * @param bool $participating
+     * @param null $since
```
Comment (xabbuh) [inline lib/Github/Api/Notification.php]: should be `string|null`

Comment (xabbuh) [inline lib/Github/Api/Notification.php]: By the way, in `markRead()` you treat `$since` as `DateTime`. Might be a good idea to to that here as well.

**Comment by pilot** [comment]
https://github.com/KnpLabs/php-github-api/releases/tag/1.4.3

**Pull request merged**